helm/templates/deployment.yaml

```diff
@@ -31,8 +31,22 @@ spec:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
       volumes:
         - name: sql-exporter
+          {{- if eq .Values.config.from.kind "ConfigMap" }}
+          configMap:
+            name: {{ .Values.config.from.name }}
+            items:
+              - key: {{ .Values.config.from.key }}
+                path: "sql_exporter.yml"
+          {{- else if eq .Values.config.from.kind "Secret" }}
+          secret:
+            secretName: {{ .Values.config.from.name }}
+            items:
+              - key: {{ .Values.config.from.key }}
+                path: "sql_exporter.yml"
+          {{- else }}
```

Comment:
Hi, To me it looks like `Values.config.value` must have the highest priority, because it's something coming with a package, and external source should be used when the value is not provided. 
```
if config.value 
  then secret from the chart
else 
  if kind == secret 
    use external secret 
  else if kind == configmap
    use external configmap
``` 
And then the same applies to the `secret.configuration.yaml`, it's created if value is provided.

Or don't you think so?

Reply:
Yes, good idea to look at `config.value` instead of `config.from`. But how do we override values when using an existing secret?

Reply:
Hm, I guess you can set it like this:
```yaml
config:
  from: 
   kind: ...
  values: {}
```
But it doesn't look nice and user-friendly to me, so yeah, maybe it makes more sense to check for the `from.kind` in that case.
Maybe @burningalchemist has an idea about that

Reply:
Ok, you can do it with `null` I guess, not `{}`, but I still don't like it

I've checked like this: 
```yaml
# template 
apiVersion: v1
kind: Service
metadata:
  {{- if .Values.check }}
  name: check
  {{- else }}
  name: not-check
  {{- end }}
  labels:
    {{- include "test.labels" . | nindent 4 }}
spec:
  type: ClusterIP
  ports:
    - port: http
      targetPort: http
      protocol: TCP
      name: http
  selector:
    {{- include "test.selectorLabels" . | nindent 4 }}
```

default values: 
```yaml
check: 
  check: check:
```

```console
$  helm template .
---
# Source: test/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: check
  labels:
    helm.sh/chart: test-0.1.0
    app.kubernetes.io/name: test
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "1.16.0"
    app.kubernetes.io/managed-by: Helm
spec:
  type: ClusterIP
  ports:
    - port: http
      targetPort: http
      protocol: TCP
      name: http
  selector:
    app.kubernetes.io/name: test
    app.kubernetes.io/instance: release-name
```

another value file: values-new.yaml
```console
$ cat values-new.yaml
check: null
$ helm template . -f values-new.yaml
---
# Source: test/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: not-check
  labels:
    helm.sh/chart: test-0.1.0
    app.kubernetes.io/name: test
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "1.16.0"
    app.kubernetes.io/managed-by: Helm
spec:
  type: ClusterIP
  ports:
    - port: http
      targetPort: http
      protocol: TCP
      name: http
  selector:
    app.kubernetes.io/name: test
    app.kubernetes.io/instance: release-name
```

Reply:
So it either should be a `_helper` function, or we just leave it like that. And though I've suggested a change, after thinking about it, I guess, that it's better if your solution stays.